app/src/main/java/org/example/app/repository/DemoRepository.java

```diff
@@ -0,0 +1,12 @@
+package org.example.app.repository;
+
+import org.example.app.entity.Demo;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.Optional;
+
+@Repository
+public interface DemoRepository extends CrudRepository<Demo, Long> {
+    Optional<Demo> findById(Long id);

```

Comment:
you don't need to overwrite this method. It already exists under the `CrudRepository` interface